### PR TITLE
Moving hardcoded enviroment var to bundler level

### PR DESCRIPTION
**Pull request by AshayKing**
ReadOnlyContext ro.skip is environment variable, moved it config level

### CODE REVIEW

File: src/main/java/io/appform/dropwizard/sharding/config/ShardedHibernateFactory.java [hunk 1]
```diff
@@ -43,4 +43,5 @@ public class ShardedHibernateFactory {
 
     @Valid
     private BlacklistConfig blacklist;
+
```
Comment (santanusinha) [inline src/main/java/io/appform/dropwizard/sharding/config/ShardedHibernateFactory.java:46]: Why not add `CustomDatabaseConfig` here with an member called `options`? This will make it configurable at startup instead of needing code changes

File: src/main/java/io/appform/dropwizard/sharding/config/CustomDatabaseConfig.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class CustomDatabaseConfig {
```
Comment (santanusinha) [inline src/main/java/io/appform/dropwizard/sharding/config/CustomDatabaseConfig.java:12]: Rename to `ShardingBundleOptions`

File: src/main/java/io/appform/dropwizard/sharding/dao/CacheableLookupDao.java [hunk 3]
```diff
@@ -43,8 +44,9 @@
 
     public CacheableLookupDao(List<SessionFactory> sessionFactories,
                               Class<T> entityClass,
-                              ShardCalculator<String> shardCalculator, LookupCache<T> cache) {
-        super(sessionFactories, entityClass, shardCalculator);
+                              ShardCalculator<String> shardCalculator, LookupCache<T> cache,
```
Comment (santanusinha) [inline src/main/java/io/appform/dropwizard/sharding/dao/CacheableLookupDao.java:47]: One param per line

File: src/main/java/io/appform/dropwizard/sharding/dao/LookupDao.java [hunk 4]
```diff
@@ -306,17 +307,15 @@ public LockedContext<T> lockAndGetExecutor(String id) {
...
 
     public ReadOnlyContext<T> readOnlyExecutor(String id, Supplier<Boolean> entityPopulator) {
         int shardId = shardCalculator.shardId(id);
         LookupDaoPriv dao = daos.get(shardId);
-        return new ReadOnlyContext<>(shardId,
-                                     dao.sessionFactory,
-                                     key -> dao.getLocked(key, LockMode.NONE),
-                                     entityPopulator,
-                                     id);
+        return new ReadOnlyContext<>(shardId, dao.sessionFactory, key -> dao.getLocked(key, LockMode.NONE),
```
Comment (santanusinha) [inline src/main/java/io/appform/dropwizard/sharding/dao/LookupDao.java:317]: One param per line

File: src/main/java/io/appform/dropwizard/sharding/dao/LookupDao.java [hunk 5]
```diff
@@ -410,110 +409,4 @@ protected Field getKeyField() {
         return this.keyField;
     }
 
-    @Getter
```
Comment (santanusinha) [inline src/main/java/io/appform/dropwizard/sharding/dao/LookupDao.java:413]: Revert unrelated change

File: src/main/java/io/appform/dropwizard/sharding/dao/RelationalDao.java [hunk 6]
```diff
@@ -251,7 +247,7 @@ <U> boolean update(LockedContext<U> context,
         }
     }
 
-    <U> List<T> select(LookupDao.ReadOnlyContext<U> context, DetachedCriteria criteria, int first, int numResults) throws Exception {
+    <U> List<T> select(ReadOnlyContext<U> context, DetachedCriteria criteria, int first, int numResults) throws Exception {
```
Comment (santanusinha) [inline src/main/java/io/appform/dropwizard/sharding/dao/RelationalDao.java:250]: Revert unrelated change

File: src/main/java/io/appform/dropwizard/sharding/dao/ReadOnlyContext.java [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+
+@Getter
+public class ReadOnlyContext<T> {
```
Comment (santanusinha) [inline src/main/java/io/appform/dropwizard/sharding/dao/ReadOnlyContext.java:19]: Revert unrelated change

**Review by santanusinha — CHANGES_REQUESTED** [review]

File: src/main/java/io/appform/dropwizard/sharding/config/ShardedHibernateFactory.java (hunk 1 above)
Comment (AshayKing) [reply src/main/java/io/appform/dropwizard/sharding/config/ShardedHibernateFactory.java:46]: added shardingOptions field in this bean. kept default as false

File: src/main/java/io/appform/dropwizard/sharding/config/CustomDatabaseConfig.java (hunk 2 above)
Comment (AshayKing) [reply src/main/java/io/appform/dropwizard/sharding/config/CustomDatabaseConfig.java]: done

File: src/main/java/io/appform/dropwizard/sharding/dao/CacheableLookupDao.java (hunk 3 above)
Comment (AshayKing) [reply src/main/java/io/appform/dropwizard/sharding/dao/CacheableLookupDao.java]: done

File: src/main/java/io/appform/dropwizard/sharding/dao/LookupDao.java (hunk 4 above)
Comment (AshayKing) [reply src/main/java/io/appform/dropwizard/sharding/dao/LookupDao.java]: done

File: src/main/java/io/appform/dropwizard/sharding/dao/LookupDao.java (hunk 5 above)
Comment (AshayKing) [reply src/main/java/io/appform/dropwizard/sharding/dao/LookupDao.java]: reverted...will move these to second mr

File: src/main/java/io/appform/dropwizard/sharding/dao/RelationalDao.java (hunk 6 above)
Comment (AshayKing) [reply src/main/java/io/appform/dropwizard/sharding/dao/RelationalDao.java]: done

File: src/main/java/io/appform/dropwizard/sharding/dao/ReadOnlyContext.java (hunk 7 above)
Comment (AshayKing) [reply src/main/java/io/appform/dropwizard/sharding/dao/ReadOnlyContext.java]: done

File: src/main/java/io/appform/dropwizard/sharding/DBShardingBundleBase.java [hunk 8]
```diff
@@ -184,31 +188,36 @@ protected ShardBlacklistingStore getBlacklistingStore() {
         return new InMemoryLocalShardBlacklistingStore();
     }
 
+    protected ShardingBundleOptions getShardingOptions() {
+        return new ShardingBundleOptions();
```
Comment (santanusinha) [inline src/main/java/io/appform/dropwizard/sharding/DBShardingBundleBase.java:192]: This should come from config. If not present default it to new object. Can take config as param to this method

**Review by santanusinha — CHANGES_REQUESTED** [review]

File: src/main/java/io/appform/dropwizard/sharding/DBShardingBundleBase.java (hunk 8 above)
Comment (AshayKing) [reply src/main/java/io/appform/dropwizard/sharding/DBShardingBundleBase.java]: done & added default object

**Review by santanusinha — APPROVED** [review]
